hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java

```diff
@@ -296,8 +296,10 @@ private void buildContainerSet() {
     ContainerInspectorUtil.load();
     while (volumeSetIterator.hasNext()) {
       StorageVolume volume = volumeSetIterator.next();
-      Thread thread = new Thread(new ContainerReader(volumeSet,
-          (HddsVolume) volume, containerSet, config, true));
+      ContainerReader containerReader = new ContainerReader(volumeSet,
+          (HddsVolume) volume, containerSet, config, true);
+      Thread thread = new Thread(containerReader,
+          datanodeDetails.threadNamePrefix());
```

Comment:
Each thread should have some unique ID in addition to the prefix.  I think we can achieve that most easily by creating a thread factory outside of the loop:

```
    String threadNamePrefix = datanodeDetails.threadNamePrefix();
    ThreadFactory threadFactory = new ThreadFactoryBuilder()
        .setDaemon(true)
        .setNameFormat(threadNamePrefix + "ContainerReader-%d")
        .build();
```

then use it to create the threads in the loop:

```
      Thread thread = threadFactory.newThread(containerReader);
```

Reply:
Thank you @adoroszlai I've changed it. Please take another look